### PR TITLE
Build script investigation

### PR DESCRIPTION
**Pull request by maxwelldb**
PR for large book master.adoc + docinfo.xml creation. 

FYI @pneedle-rh

### CODE REVIEW

File: build_for_portal.py [hunk 1]
```diff
@@ -6,36 +6,49 @@
...
 COMMENT_CONTENT_RE = re.compile(r"^^////$.*?^////$", re.M | re.DOTALL)
-TAG_CONTENT_RE = re.compile(r"//\s+tag::(.*?)\[\].*?// end::(.*?)\[\]", re.M | re.DOTALL)
+TAG_CONTENT_RE = re.compile(
+    r"//\s+tag::(.*?)\[\].*?// end::(.*?)\[\]", re.M | re.DOTALL
+)
 CMP_IGNORE_FILES = [".git", ".gitignore", "README.md", "build.cfg"]
-DEVNULL = open(os.devnull, 'wb')
-
+DEVNULL = open(os.devnull, "wb")
+LIST_OF_HUGE_BOOKS: list[str] = ["Installing", "API reference"]
```
Comment (maxwelldb) [inline build_for_portal.py:51]: Important.

**Comment by pneedle-rh** [comment]
@maxwelldb and @kalexand-rh hi! This is great work, Max!

I tested the updates that are related to the large book handling. I ran some tests in an interactive Python shell and then by calling the updated `build_for_portal.py` script from the 4.11 sync script with the `git push` commented out.

All looks good to me aside from not seeing any `docinfo.xml` files generated in the subdirectories for the installing book. The results were instead as follows:

```
$ find 4.11_temp/doc-4.11/installing/ -iname "master.adoc"
4.11_temp/doc-4.11/installing/disconnected_install/master.adoc
4.11_temp/doc-4.11/installing/install_config/master.adoc
4.11_temp/doc-4.11/installing/installing_alibaba/master.adoc
4.11_temp/doc-4.11/installing/installing_aws/master.adoc
4.11_temp/doc-4.11/installing/installing_azure/master.adoc
4.11_temp/doc-4.11/installing/installing_azure_stack_hub/master.adoc
4.11_temp/doc-4.11/installing/installing_bare_metal/master.adoc
4.11_temp/doc-4.11/installing/installing_bare_metal_ipi/master.adoc
4.11_temp/doc-4.11/installing/installing_gcp/master.adoc
4.11_temp/doc-4.11/installing/installing_ibm_cloud/master.adoc
4.11_temp/doc-4.11/installing/installing_ibm_cloud_public/master.adoc
4.11_temp/doc-4.11/installing/installing_ibm_power/master.adoc
4.11_temp/doc-4.11/installing/installing_ibm_z/master.adoc
4.11_temp/doc-4.11/installing/installing_nutanix/master.adoc
4.11_temp/doc-4.11/installing/installing_on_prem_assisted/master.adoc
4.11_temp/doc-4.11/installing/installing_openstack/master.adoc
4.11_temp/doc-4.11/installing/installing_platform_agnostic/master.adoc
4.11_temp/doc-4.11/installing/installing_rhv/master.adoc
4.11_temp/doc-4.11/installing/installing_sno/master.adoc
4.11_temp/doc-4.11/installing/installing_vmc/master.adoc
4.11_temp/doc-4.11/installing/installing_vsphere/master.adoc
4.11_temp/doc-4.11/installing/master.adoc

pneedle@laptop sync_scripts $ find 4.11_temp/doc-4.11/installing/ -iname "docinfo.xml"
4.11_temp/doc-4.11/installing/docinfo.xml
```

Is there supposed to be a `docinfo.xml` file in each of the subdirectories too, after the script is run, or have I misunderstood?

I have made detailed notes from my testing and will send you a link to that outside of this PR.

**Comment by maxwelldb** [comment]
@pneedle-rh https://github.com/openshift/openshift-docs/pull/52095/commits/ac940691baf793d08b5dd8a9423e94110e92a5fb Seems to fix the docinfo.xml title issue.

**Comment by pneedle-rh** [comment]
@maxwelldb hi! I tested this again today, using the updated version of the `build_for_portal.py` script. I get the same result, with no `docinfo.xml` files in the Installing book subdirectories.

Below I have outlined the explicit steps that I ran in my testing.

### Steps to reproduce the problem

1. I made a local clone of your fork and checked out to the `build-script-investigation` feature branch in the local clone.

2. I copied the `build_for_portal.py` script from that feature branch to `/var/tmp/build_for_portal.py`.

3. I fetched and rebased the `stage` branch in my local clone of the `doc-4.11` GitLab branch.

4. I copied `sync.sh` from that `stage` branch to `/var/tmp/sync_scripts` and ran `chmod 755 sync.sh` to update the permissions to make it executable.

5. I made two changes to `/var/tmp/sync_scripts/sync.sh`.

a. Firstly, I added the following line to ensure that your `build_for_portal.py` script is copied to the temporary `openshift-docs` local clone that is cloned by the sync script. This copy occurs after the checkout to `enterprise-${version}` and before the `build_for_portal.py` script is called:

`56	cp /var/tmp/build_for_portal.py .`

b. I also commented out the `git push`:

`85	#git push origin stage`

6. Then, I ran the sync script:

```
$ pwd
/var/tmp/sync_scripts

$ ./sync.sh 4.11
```

### Result

The output of the two find commands were the same as before:

```
$ find 4.11_temp/doc-4.11/installing/ -iname "master.adoc"
4.11_temp/doc-4.11/installing/disconnected_install/master.adoc
4.11_temp/doc-4.11/installing/install_config/master.adoc
4.11_temp/doc-4.11/installing/installing_alibaba/master.adoc
4.11_temp/doc-4.11/installing/installing_aws/master.adoc
4.11_temp/doc-4.11/installing/installing_azure/master.adoc
4.11_temp/doc-4.11/installing/installing_azure_stack_hub/master.adoc
4.11_temp/doc-4.11/installing/installing_bare_metal/master.adoc
4.11_temp/doc-4.11/installing/installing_bare_metal_ipi/master.adoc
4.11_temp/doc-4.11/installing/installing_gcp/master.adoc
4.11_temp/doc-4.11/installing/installing_ibm_cloud/master.adoc
4.11_temp/doc-4.11/installing/installing_ibm_cloud_public/master.adoc
4.11_temp/doc-4.11/installing/installing_ibm_power/master.adoc
4.11_temp/doc-4.11/installing/installing_ibm_z/master.adoc
4.11_temp/doc-4.11/installing/installing_nutanix/master.adoc
4.11_temp/doc-4.11/installing/installing_on_prem_assisted/master.adoc
4.11_temp/doc-4.11/installing/installing_openstack/master.adoc
4.11_temp/doc-4.11/installing/installing_platform_agnostic/master.adoc
4.11_temp/doc-4.11/installing/installing_rhv/master.adoc
4.11_temp/doc-4.11/installing/installing_sno/master.adoc
4.11_temp/doc-4.11/installing/installing_vmc/master.adoc
4.11_temp/doc-4.11/installing/installing_vsphere/master.adoc
4.11_temp/doc-4.11/installing/master.adoc

$ find 4.11_temp/doc-4.11/installing/ -iname "docinfo.xml"
4.11_temp/doc-4.11/installing/docinfo.xml
```

Is this an issue in the way that I'm testing this? How are you running your testing?

**Comment by maxwelldb** [comment]
@pneedle-rh Can you confirm that running the build script on its own is generating the correct docinfo.xml files?

**Comment by pneedle-rh** [comment]
@maxwelldb the script has generated the relevant files in the temporary `openshift-docs` repository in the `drupal-build` directory, so I assume from this that the build script is generating the files:

```
$ find openshift-docs/drupal-build/openshift-enterprise/installing/ -iname "docinfo.xml"
openshift-docs/drupal-build/openshift-enterprise/installing/docinfo.xml
openshift-docs/drupal-build/openshift-enterprise/installing/disconnected_install/docinfo.xml
openshift-docs/drupal-build/openshift-enterprise/installing/installing_alibaba/docinfo.xml
openshift-docs/drupal-build/openshift-enterprise/installing/installing_aws/docinfo.xml
openshift-docs/drupal-build/openshift-enterprise/installing/installing_azure/docinfo.xml
openshift-docs/drupal-build/openshift-enterprise/installing/installing_azure_stack_hub/docinfo.xml
openshift-docs/drupal-build/openshift-enterprise/installing/installing_gcp/docinfo.xml
openshift-docs/drupal-build/openshift-enterprise/installing/installing_ibm_cloud_public/docinfo.xml
openshift-docs/drupal-build/openshift-enterprise/installing/installing_nutanix/docinfo.xml
openshift-docs/drupal-build/openshift-enterprise/installing/installing_bare_metal/docinfo.xml
openshift-docs/drupal-build/openshift-enterprise/installing/installing_on_prem_assisted/docinfo.xml
openshift-docs/drupal-build/openshift-enterprise/installing/installing_sno/docinfo.xml
openshift-docs/drupal-build/openshift-enterprise/installing/installing_bare_metal_ipi/docinfo.xml
openshift-docs/drupal-build/openshift-enterprise/installing/installing_ibm_cloud/docinfo.xml
openshift-docs/drupal-build/openshift-enterprise/installing/installing_ibm_z/docinfo.xml
openshift-docs/drupal-build/openshift-enterprise/installing/installing_ibm_power/docinfo.xml
openshift-docs/drupal-build/openshift-enterprise/installing/installing_openstack/docinfo.xml
openshift-docs/drupal-build/openshift-enterprise/installing/installing_rhv/docinfo.xml
openshift-docs/drupal-build/openshift-enterprise/installing/installing_vsphere/docinfo.xml
openshift-docs/drupal-build/openshift-enterprise/installing/installing_vmc/docinfo.xml
openshift-docs/drupal-build/openshift-enterprise/installing/installing_platform_agnostic/docinfo.xml
openshift-docs/drupal-build/openshift-enterprise/installing/install_config/docinfo.xml
```

For some reason those are not showing in the doc-4.11 local clone. I think it might just be that the `sync.sh` script is not rsyncing `docinfo.xml` over from subdirectories.

**Comment by maxwelldb** [comment]
@pneedle-rh I assumed it was a sync script issue, but didn't have time this AM to check that out. 

That section of `sync.sh` looks highly relevant. 😬

**Comment by maxwelldb** [comment]
@pneedle-rh So, after chatting with Kathryn, it seems like the script-generated docinfo.xml files aren't used at all. It's a manual operation, which, while not ideal, means that I think this could be called 'done' as an MVP. Do you concur? 

I'd like to improve this process, as the status quo seems like a pain, but that would be for another sprint. 

Awaiting an ack on this from @kalexand-rh.

**Comment by kalexand-rh** [comment]
I agree both that the docinfo.xml creation is currently manual and that more improvements could happen in a future sprint. The docinfo.xml files contain values that are custom for each directory, so it would require more extensive changes to automate their creation.

**Comment by pneedle-rh** [comment]
@maxwelldb and @kalexand-rh hi! If the required result is for the `master.adoc` files from the large book subdirectories to be generated, copied over to the local GitLab clone and pushed to the mainstream branch, then this PR LGTM!

Great work, thanks for involving me!

**Pull request merged**